blog_post.md

```diff
@@ -1,52 +1,290 @@
+kind: RabbitmqCluster
+metadata:
+  name: rabbitmq
+  annotations:
+    service.binding: path={.status.binding.name},objectType=Secret,elementType=map
+spec:
+  service:
+    type: ClusterIP
+```
+
+To make this process easier, you can deploy all of these (that is, `producer`,
+`consumer`, and our rabbitmq cluster) with the following:
+```bash
+kubectl apply -f https://raw.githubusercontent.com/sadlerap/sbo-rabbitmq-sample/master/jobs.yaml
+```
+
+NOTE: Look closely at the annotation we've added.  This is how SBO picks up the
+information it needs to successfully perform a binding.  If you're interested
+in how to read this annotation, check out our
+[documentation](https://redhat-developer.github.io/service-binding-operator/userguide/exposing-binding-data/adding-annotation.html).
+
+Now, if you inspect our container logs for `consumer`, you'll see something
+similar to this:
+```
+$ kubectl logs consumer-deployment-f877cffb6-p9sks
+Error:
+   0: $RABBITMQCLUSTER_HOST not defined
+
+Location:
+   src/consumer.rs:16
+
+Backtrace omitted.
+Run with RUST_BACKTRACE=1 environment variable to display it.
+Run with RUST_BACKTRACE=full to include source snippets.
+```
+
+`producer` throws a similar error.  This is because we haven't bound our
+RabbitMQ cluster to either `producer` and `consumer`, so neither of our
+containers know where to find it.  Let's fix that.
 
-In normal circumstances (read: not running on kubernetes), we would need to tell
-both `producer` and `consumer` how to connect to a rabbitmq instance.  This
-means distributing to these services the following information:
+## Binding things together
 
-- Hostname/Port
-- Authentication credentials (such as username & password)
+If we were not using SBO, we would need to tell both `producer` and `consumer`
+how to connect to a rabbitmq instance.  This means distributing at minimum the
+following information to these services:
 
-In summary, for our example, we want the following:
+- The hostname of the RabbitMQ instance
+- The port that the RabbitMQ instance is listening on
+- Authentication credentials (such as username and password)
 
-- An operator-managed RabbitMQ cluster running on kubernetes (we will use
-  https://github.com/rabbitmq/cluster-operator for this demonstration)
-- Our `producer` and `consumer` also running on kubernetes
+This would require us to expose our secrets to our applications, either by
+having these applications directly fetch that information from kubernetes' API
+or by injecting that information into our applications ourselves.  Both of
+these methods are rather intrusive into our applications, and it stands to
+reason that this process could be automated.
 
-# Connecting the hard way
+First, we need to allow SBO to make a binding against our RabbitMQ cluster by
+adding a ClusterRole.  For security reasons, SBO doesn't have permissions to
+make a binding against something it doesn't know about; we can remedy this by
+adding a ClusterRole like so:
```

Comment:
this is not needed for upcoming SBO 1.0 since we support RabbitMq operator out-of-the-box

Reply:
I'd still like to mention something about RBAC rules, since they can be a source of issues.  I'll rewrite this to refer to the registry for why we don't have to do it here.